guava/src/test/java/com/fasterxml/jackson/datatype/guava/RangeDeserializer188Test.java

```diff
@@ -28,52 +33,89 @@ public Range<T> toRange() {
-        assertEquals("{\"r\":\"(-∞..+∞)\"}",
-                MAPPER.writeValueAsString(new Stringified<>(unboundedRange)));
-
-        // Test open range serialization with Duration
-        Range<Duration> openDurationRange = RangeFactory.open(Duration.ofMinutes(30), Duration.ofMinutes(60));
-        assertEquals("{\"r\":\"(PT30M..PT1H)\"}",
-                MAPPER.writeValueAsString(new Stringified<>(openDurationRange)));
-
-        // Test closed range serialization with LocalDate
-        Range<LocalDate> closedDateRange = RangeFactory.closed(LocalDate.parse("2023-01-01"), LocalDate.parse("2023-01-31"));
-        assertEquals("{\"r\":\"[2023-01-01..2023-01-31]\"}",
-                MAPPER.writeValueAsString(new Stringified<>(closedDateRange)));
-
-        // Test open-closed range serialization with String
-        Range<String> openClosedStringRange = RangeFactory.openClosed("abc", "def");
-        assertEquals("{\"r\":\"(abc..def]\"}",
-                MAPPER.writeValueAsString(new Stringified<>(openClosedStringRange)));
+        testRangeSerialization(openRange, "(1..10)");
+        testRangeSerialization(closedRange, "[5..15]");
+        testRangeSerialization(openClosedRange, "(10..20]");
+        testRangeSerialization(closedOpenRange, "[25..30)");
+        testRangeSerialization(singletonRange, "[42..42]");
+        testRangeSerialization(unboundedRange, "(-∞..+∞)");
+        testRangeSerialization(openDurationRange, "(PT30M..PT1H)");
+        testRangeSerialization(closedDateRange, "[2023-01-01..2023-01-31]");
+        testRangeSerialization(openClosedStringRange, "(abc..def]");
+        testRangeSerialization(leftBoundedRange, "(-∞..15)");
+        testRangeSerialization(rightBoundedRange, "[10..+∞)");
+    }
+
+    // [dataformats-collections#135]
+    public void testRangeDeserializationFromBracketNotation() throws Exception
+    {
+        testRangeDeserialization("{\"r\":\"(1..10)\"}", openRange);
+        testRangeDeserialization("{\"r\":\"[5..15]\"}", closedRange);
+        testRangeDeserialization("{\"r\":\"(10..20]\"}", openClosedRange);
+        testRangeDeserialization("{\"r\":\"[25..30)\"}", closedOpenRange);
+        testRangeDeserialization("{\"r\":\"[42..42]\"}", singletonRange);
+        testRangeDeserialization("{\"r\":\"(-∞..+∞)\"}", unboundedRange);
+        testRangeDeserialization("{\"r\":\"(-∞..15)\"}", leftBoundedRange);
+        testRangeDeserialization("{\"r\":\"[10..+∞)\"}", rightBoundedRange);
+
+        // The below are failing for now. Priority is to get the minimal working deserialization.
+
+        // testRangeDeserialization("{\"r\":\"(PT30M..PT1H)\"}", openDurationRange);
+        // testRangeDeserialization("{\"r\":\"[2023-01-01..2023-01-31]\"}", closedDateRange);
+        // testRangeDeserialization("{\"r\":\"(abc..def]\"}", openClosedStringRange);
+    }
+
+    public void testInvalidBracketNotationRangeDeserialization() throws Exception {
```

Comment:
@cowtowncoder,

Added more invalid test cases to check if the exception thrown is correct.